rumqttd/src/server/tls.rs

```diff
@@ -227,3 +219,33 @@ impl TLSAcceptor {
         Ok(TLSAcceptor::Rustls { acceptor })
     }
 }
+
+#[cfg(feature = "use-rustls")]
+/// Get the first private key in a PEM file
+fn first_private_key_in_pemfile(key_path: &String) -> Result<PrivateKey, Error> {
+    // Get private key
+    let key_file = File::open(key_path);
+    let key_file = key_file.map_err(|_| Error::ServerKeyNotFound(key_path.clone()))?;
+
+    // Let's iterate over the keys with read_one/read_all
+    let rd = &mut BufReader::new(key_file);
+
+    loop {
+        match rustls_pemfile::read_one(rd) {
+            Ok(maybe_inner) => match maybe_inner {
+                None => {
+                    error!("No private key found in {:?}", key_path);
+                    return Err(Error::InvalidServerKey(key_path.clone()));
+                }
+                Some(Item::ECKey(key)) => return Ok(PrivateKey(key)),
+                Some(Item::RSAKey(key)) => return Ok(PrivateKey(key)),
+                Some(Item::PKCS8Key(key)) => return Ok(PrivateKey(key)),
```

Comment:
will it be better to use `|` like `Some(Item::ECKey | Item::RSAKey | ..)` ? 
ps: only do this if it doesn't compromise readability!

nit: can we move `None` below this? as it is less intended.

Reply:
Done, now looks like this:


```rust
    loop {
        match rustls_pemfile::read_one(rd) {
            Ok(maybe_inner) => match maybe_inner {
                Some(item) => match item {
                    // If we find a private key, return it
                    Item::ECKey(key) | Item::RSAKey(key) | Item::PKCS8Key(key) => {
                        return Ok(PrivateKey(key))
                    }
                    // Otherwise, continue
                    _ => {}
                },
                None => {
                    error!("No private key found in {:?}", key_path);
                    return Err(Error::InvalidServerKey(key_path.clone()));
                }
            },
            Err(err) => {
                error!("Error reading key file: {:?}", err);
                return Err(Error::InvalidServerKey(key_path.clone()));
            }
        }
    }
```

Reply:
we should also note that this is giving preference like `EC` > `RSA` > `PKCS8`! maybe add it as a comment?